src/box/errcode.h

```diff
@@ -115,286 +115,286 @@ struct errcode_record {
-	_(ER_VIEW_IS_RO, 113,			"View '%s' is read-only") \
+	_(ER_UNSUPPORTED_PRIV, 98,		"Unsupported privilege", "object_type", STRING, "privilege", STRING) \
+	_(ER_LOAD_FUNCTION, 99,			"Failed to dynamically load function", "function", STRING, "details", STRING) \
+	_(ER_FUNCTION_LANGUAGE, 100,		"Unsupported function language", "language", STRING, "function", STRING) \
+	_(ER_RTREE_RECT, 101,			"Coordinates number does not make a point or rectangle", "object", STRING, "dimension", UINT) \
+	_(ER_PROC_C, 102,			"Error in C function") \
+	/* ER_UNKNOWN_RTREE_INDEX_DISTANCE_TYPE, 103, Unused */ \
+	_(ER_PROTOCOL, 104,			"Protocol error", "details", STRING) \
+	/* ER_UPSERT_UNIQUE_SECONDARY_KEY, 105,	Unused */ \
+	_(ER_WRONG_INDEX_RECORD, 106,		"Wrong record in _index space", "actual", STRING, "expected", STRING) \
+	_(ER_WRONG_INDEX_PARTS, 107,		"Wrong index part", "partno", UINT, "details", STRING) \
+	_(ER_WRONG_INDEX_OPTIONS, 108,		"Wrong index options", "details", STRING) \
+	_(ER_WRONG_SCHEMA_VERSION, 109,		"Request schema version does not match database schema version", "actual", ULLONG, "expected", ULLONG) \
+	_(ER_MEMTX_MAX_TUPLE_SIZE, 110,		"Failed to allocate tuple: tuple is too large", "tuple_size", UINT) \
+	_(ER_WRONG_SPACE_OPTIONS, 111,		"Wrong space options") \
+	_(ER_UNSUPPORTED_INDEX_FEATURE, 112,	"Unsupported index feature") \
+	_(ER_VIEW_IS_RO, 113,			"View is read-only", "space", STRING) \
 	_(ER_NO_TRANSACTION, 114,		"No active transaction") \
-	_(ER_SYSTEM, 115,			"%s") \
+	_(ER_SYSTEM, 115,			"System error") \
 	_(ER_LOADING, 116,			"Instance bootstrap hasn't finished yet") \
 	_(ER_CONNECTION_TO_SELF, 117,		"Connection to self") \
-	_(ER_KEY_PART_IS_TOO_LONG, 118,		"Key part is too long: %u of %u bytes") \
-	_(ER_COMPRESSION, 119,			"Compression error: %s") \
+	/* ER_KEY_PART_IS_TOO_LONG, 118, Unused */ \
+	_(ER_COMPRESSION, 119,			"Compression error", "details", STRING) \
 	_(ER_CHECKPOINT_IN_PROGRESS, 120,	"Snapshot is already in progress") \
 	_(ER_SUB_STMT_MAX, 121,			"Can not execute a nested statement: nesting limit reached") \
 	_(ER_COMMIT_IN_SUB_STMT, 122,		"Can not commit transaction in a nested statement") \
 	_(ER_ROLLBACK_IN_SUB_STMT, 123,		"Rollback called in a nested statement") \
-	_(ER_DECOMPRESSION, 124,		"Decompression error: %s") \
-	_(ER_INVALID_XLOG_TYPE, 125,		"Invalid xlog type: expected %s, got %s") \
-	_(ER_ALREADY_RUNNING, 126,		"Failed to lock WAL directory %s and hot_standby mode is off") \
-	_(ER_INDEX_FIELD_COUNT_LIMIT, 127,	"Indexed field count limit reached: %d indexed fields") \
-	_(ER_LOCAL_INSTANCE_ID_IS_READ_ONLY, 128, "The local instance id %u is read-only") \
+	_(ER_DECOMPRESSION, 124,		"Decompression error", "details", STRING) \
+	_(ER_INVALID_XLOG_TYPE, 125,		"Invalid xlog type", "expected", STRING, "actual", STRING) \
+	_(ER_ALREADY_RUNNING, 126,		"Failed to lock WAL directory and hot_standby mode is off", "directory", STRING) \
```

Comment:
I don't think we need to specify the directory here - it's known from the config.

Reply:
Changed to `"path"` as discussed f2f (let's keep it as we have config values in payload for other errors).

```diff
+       _(ER_ALREADY_RUNNING, 126,              "Failed to lock WAL directory and hot_standby mode is off", "path", STRING) \
```